### DISCUSSION TITLE
Figuring Things Out -- FF vs OOB Gotek -- Having Issues

### DISCUSSION THREAD

**Discussion by Raven2k360** [keirf/flashfloppy#675] [Q&A]
After successfully flashing a Gotek Unit with the FF firmware yesterday, I still can't get it to access a USB drive, no matter what formatting I use.  When I plug a stick in, it's internal LED will flash as it's spinning up, but after it goes quiet and I tell the Host to access the floppy drive, the Gotek's LED will flash for a few seconds, then go out - the USB stick's LED won't light up at all, so it's not even trying to access it as far as I can tell.  I've tried several different formatting options for the drive, as well as dumping the FF.CFG file onto the stick as well, and I've also tried several different jumper configurations on the Gotek itself (S0, S1, or M0, as well as JC+ each of them).

Conversely, I can get a raw, out of the box Gotek unit (the other one I ordered) to read my USB stick, but ONLY if I format it to a 1.44MB disk - so the USB will ONLY show 1.44MB capacity.  This seems highly illogical to me to do things this way, considering other options don't have this restriction/limitation.

As another comparison, the Richpeace units (configured for IBM-PC) we have in some boxes (Inficon IC-5 Deposition Controllers) here at work, while they won't read my USB sticks for some reason (PNY 16GB USB2.0 units), will read other sticks we've successfully backed up to - two are SanDisk Cruzer 64GB sticks and one is a Cruzer Mini 256MB stick, and the host box will read their full capacity.  I'm also going to reach out to Richpeace and/or PLR here in the States (they sell/rebadge the Richpeace units) about this issue.

Back to the Gotek/FF issues though - is there any way to get the Gotek drives to read a USB stick without formatting down to 1.44MB?  And why won't the module with the FF firmware access the stick at all?

**Comment by keirf** [comment]
What does the Gotek display say after you insert the usb drive?

**Reply by Raven2k360** [reply]
I have two models -- two with no display (one is flashed with the FF firmware) and one with the 3-LED display (ordered by a former co-worker).  The one with the 3-LED display is set to '000'.

**Reply by keirf** [reply]
000 is good. That means some image file is mounted. What files do you have on your usb drive? 

Displayless models are hard to troubleshoot. Who knows what mode they are in.

**Reply by keirf** [reply]
After some seconds the display should change to 0_ which is the track display: it means cylinder 0, lower head.

**Reply by keirf** [reply]
Is the gotek sfrkc30.at2 model and if so does it actually have a JC jumper location? Most don't.

**Reply by Raven2k360** [reply]
All 3 units are running that revision of board.  The displayless models have a JC set of pins and silkscreen label, but the LED-display model has the pins, but no silkscreen label.  When shining a light through the board from the back, there are also noticeable differences around the header - on the unit without the JC label, it looks like both pins are tied to the J5 and JA pins on the same plane, whereas the displayless models have the inboard pin separated - it still doesn't look like the pin goes anywhere though...hard to tell without a schematic.  I can take photos if needed.

When the LED-Display model accesses the USB stick, there is no display change - it remains at '000', until I manually index it to another 'drive'.  We've also successfully used this unit and a stick formatted into multiple floppy images/folders, so the indexing function works as it should.  As a side note, when using the stick as a single 1.44MB disk, if I index off of '000', it will not access the stick until I index back.

**Reply by keirf** [reply]
No need. If it doesn't have JC on the silk layer then that jumper location doesn't exist. Concentrating on this drive, create a FF.CFG file containing only the line `interface=ibmpc`

Now jumper location S1 on the Gotek. It is the most likely drive select line. Also place one IMG file on the usb stick.

**Reply by keirf** [reply]
Have you had *any* other non default options in your FF.CFG file?

**Reply by keirf** [reply]
May I also ask what the display says when no usb stick is inserted.

**Reply by Raven2k360** [reply]
I'm assuming to use the FF.CFG file, I have to first flash the FF firmware to the unit - I'm a bit reticent to do this on that particular unit, especially since it hasn't helped the one that was flashed and I don't have a spare unit of that model, or a way to flash back to the stock Gotek firmware.  I can try the config file with an IMG file on the one that has been flashed however, and just leave the JC jumper off.  On a related note, is the FF.CFG file read by the firmware directly?  If so, how/when is this accomplished?  On the unit with the FF firmware, the presence of the FF.CFG file made no difference in whether the drive was accessed or not after the initial plug-in.  The FF.CFG file I used in my tests was the one from the example folder, with the only option being changed was 'interface = ibmpc'.  All other options in the file were left as-is.

The display on the LED model still showes '000' with no stick plugged in.  The Non-Display models are SFR1M44-U units, and the Display model is a SFR1M44-U100 unit.

**Reply by keirf** [reply]
Oh I see. Well it's pretty hard to diagnose a unit with no display at all so I'll tap out on that I'm afraid. 

Maybe swap the FF board into the case with the display. It's possible that you would have to solder a header onto that FF board to attach the display though. But if you're lucky the factory will have soldered a header on already.

**Reply by Raven2k360** [reply]
Yeah, the PCBs in the Non-Display units are missing the headers for the display and navigation buttons.

I did notice that when I formatted the USB stick for a single 1.44MB drive, the program I used also created a file called 'Image A.cfg', but it's blank when I open it in notepad.  I'm assuming the Gotek drives in general operate on a disk image setup, rather than accessing the drive directly?  I guess that would make sense why I can't access the flash drive's full capacity natively.  Is there no way to get the Gotek drives to do this?  There's at least one company who's drives operate this way - though they may be the only one....it seems the Floppy to USB market is rather small...lol.

I'll look into getting some pin headers and solder them to the FF-Flashed PCB - it may take a few days to get around to that though.  Thanks again for taking the time to work with me on this.

**Reply by keirf** [reply]
FlashFloppy works on IMG files within a FAT32 formatted USB drive. That means:
1. Make sure your USB drive is FAT formatted (not exFAT, not NTFS, etc)
2. Copy at least one IMG file onto the drive. Probabkly you want 1.44MB IMG files, which are analagous to 1.44MB PC HD floppy disks. 

To manage the IMG files on your Windows PC you will want a program like WinImage (shareware) or ImDisk (free). ImDisk is good because you can format a new IMG file and mount it as a drive letter on modern Windows! 

Below is an empty formatted 1.44MB IMG file to get you started.

[00.zip](https://github.com/keirf/flashfloppy/files/9221018/00.zip)
